File: app/src/main/java/me/lukeforit/spaceofaday/ui/base/Event.java
package me.lukeforit.spaceofaday.ui.base;

public class Event<T> {
    private T data;
    private boolean delivered;

    public Event(T data) {
        this.data = data;
        delivered = false;
    }

    public boolean isDelivered() {
        return delivered;
    }

    public T deliverData() {
        if (delivered) {
            return null;
        } else {
            delivered = true;
            return data;
        }
    }

    public T peekData() {
        return data;
    }
}

File: app/src/main/java/me/lukeforit/spaceofaday/ui/base/Event.kt
package me.lukeforit.spaceofaday.ui.base

class Event<T>(private val data: T) {
    var isDelivered: Boolean = false
        private set

    init {
        isDelivered = false
    }

    fun deliverData(): T? {
        if (isDelivered) {
            return null
        } else {
            isDelivered = true
            return data
        }
    }

    fun peekData(): T {
        return data
    }
}
